src/main/kotlin/com/devfloor/untitled/articlehitscount/application/ArticleHitsCountService.kt

```diff
@@ -0,0 +1,22 @@
+package com.devfloor.untitled.articlehitscount.application
+
+import com.devfloor.untitled.article.domain.Article
+import com.devfloor.untitled.articlehitscount.domain.ArticleHitsCount
+import com.devfloor.untitled.articlehitscount.domain.ArticleHitsCountRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class ArticleHitsCountService(
+    private val articleHitsCountRepository: ArticleHitsCountRepository,
+) {
+    @Transactional
+    fun increase(article: Article) {
+        var articleHitsCount = articleHitsCountRepository.findByArticleOrNull(article)
+        if (articleHitsCount == null) {
+            articleHitsCount = ArticleHitsCount(article)
+        }
+        articleHitsCount.increase()
+        articleHitsCountRepository.save(articleHitsCount)
+    }
```

Comment:
아니면 전체를 이러한 형태로 바꾸는건 어떨까요!?
조회해서 있으면 바꿔주고 더티체킹 이용하도록, 없으면 직접 저장해주는 로직으로 이렇게 짜보았는데, 문제는 이렇게 할라면 ArticleHits 생성 시점에 좋아요 값을 1로 생성하도록 해야할거같아요.!

```suggestion
    fun increase(article: Article) = articleHitsCountRepository.findByArticleOrNull(article)
        ?.run { ::increase }
        ?: articleHitsCountRepository.save(ArticleHitsCount(article))
```

Reply:
suggestion 코드 14번째 줄에서 자동으로 개행이 되어 보이네요.. 의도한 바는 아래와 같습니다 !
![image](https://user-images.githubusercontent.com/49271821/122709580-89e13880-d299-11eb-8a4b-d22b7b0f3076.png)

Reply:
increase 이후 (?. run에 해당하는) 변경된 객체를 save해야하는데 increase 자체가 setter함수로 리턴값없이 구현해놔서..
setter함수의경우 해당 변경한 객체를 리턴하도록 만들어도 되는건가요?? 대부분 자료가 void로 구현되어지길래 ..